clients/src/main/java/org/apache/kafka/common/protocol/SecurityProtocol.java

```diff
@@ -26,6 +26,8 @@
     PLAINTEXT(0, "PLAINTEXT"),
     /** SSL channel */
     SSL(1, "SSL"),
+    /** SASL authenticated, non-encrypted channel */
+    PLAINTEXTSASL(2, "PLAINTEXTSASL"),
```

Comment:
@ijuma @junrao SASL/PLAIN is typically used to refer to SASL with mechanism PLAIN. And SASL/PLAIN is usually used with SSL as transport layer. Since the protocols here are referring to the transport layer and the plain transport layer is called PLAINTEXT, it would be less confusing to have SASL_PLAINTEXT and SASL_SSL.